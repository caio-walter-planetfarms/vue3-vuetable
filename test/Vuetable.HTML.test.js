import { mount } from '@vue/test-utils'
import Vuetable from 'components/Vuetable.vue'

describe('Vuetable - HTML structure', () => {

  const mountVuetable = (fields) => mount(Vuetable, {
    propsData: {
      loadOnStart: false,
      fields
    }
  })

  it('renders HTML table', () => {
    const wrapper = mountVuetable([
      'code', 'description'
    ])

    expect(wrapper.element).toMatchSnapshot()
  })

})
