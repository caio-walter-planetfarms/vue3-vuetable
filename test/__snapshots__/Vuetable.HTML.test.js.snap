// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vuetable - HTML structure renders HTML table 1`] = `
<div
  class=""
>
  <!--v-if-->
  <div
    class="vuetable-body-wrapper"
  >
    <table
      class="vuetable ui blue selectable unstackable celled table fixed"
    >
      <colgroup>
        
        
        <col
          class="vuetable-col-code"
        />
        
        
        <col
          class="vuetable-col-description"
        />
        
        
        <!--v-if-->
      </colgroup>
      <thead>
        
        
        <tr>
          
          
          
          <th
            class="vuetable-th-code"
            id="_code"
          >
            Code
          </th>
          
          
          
          
          <th
            class="vuetable-th-description"
            id="_description"
          >
            Description
          </th>
          
          
          
          <!--v-if-->
        </tr>
        
        
      </thead>
      <tfoot>
        
        
      </tfoot>
      <tbody
        class="vuetable-body"
      >
        
        
        <tr>
          <td
            class="vuetable-empty-result"
            colspan="2"
          >
            No Data Available
          </td>
        </tr>
        
        
      </tbody>
    </table>
  </div>
</div>
`;
